src/java.base/share/classes/java/net/Socket.java

```diff
@@ -649,21 +657,21 @@ public void connect(SocketAddress endpoint, int timeout) throws IOException {
         if (!(endpoint instanceof InetSocketAddress epoint))
             throw new IllegalArgumentException("Unsupported address type");
 
+        if (epoint.isUnresolved()) {
+            UnknownHostException exception = new UnknownHostException(epoint.getHostName() + " is unresolved");
```

Comment:
In passing, if you replace this with `var uhe = new UnknownHostException(...)` then the later "throw uhe` will be clearer than `throw exception`.

Reply:
Fixed in 92ef9ebb8eff8d6972d5363bdce6b88f55c4a7c4.